src/garage/np/algos/base.py

```diff
@@ -25,4 +30,60 @@ def train(self, runner):
             The average return in last epoch cycle or None.
 
         """
-        pass
+
+    def evaluate_performance(self, itr, batch):
+        # pylint: disable=no-self-use
+        r"""Evaluate the performance of the algorithm.
+
+        Args:
+            itr (int): Iteration number.
+            batch (dict): Evaluation trajectories, representing
+                the best current performance of the algorithm, with keys:
+                * env_spec (garage.envs.EnvSpec): Specification for the
+                environment from which this data was sampled.
+                * observations (numpy.ndarray): A numpy array containing the
+                    observations for all time steps in this batch.
+                * actions (numpy.ndarray): A  numpy array containing the
+                    actions for all time steps in this batch.
+                * rewards (numpy.ndarray): A numpy array containing the
+                    rewards for all time steps in this batch.
+                * terminals (numpy.ndarray): A boolean numpy array
+                    containing the termination signals for all time steps
+                    in this batch.
+                * env_infos (dict): A dict of numpy arrays arbitrary
+                    environment state information.
+                * agent_infos (numpy.ndarray): A dict of numpy arrays
+                    arbitrary agent state information.
+                * lengths (numpy.ndarray): An integer numpy array
+                    containing the length of each trajectory in this batch.
+                * discount (float): Discount value.
```

Comment:
this should be a property of `RLAlgorithm`, i.e. `self.discount_factor`

Reply:
It is using self.discount from algorithms. But since base.py is an interface so I passed it.

Reply:
Can you add this as a constructor argument + instance variable + read-only property of RLAlgorithm?

Reply:
Okay, leave it for now. Let's focus on getting this merged.